### PR TITLE
Let pipeline-compiler export step types

### PR DESCRIPTION
**Pull request by 6543**
groundwork to address  #269 (missing part are UI changes)

### CODE REVIEW

File: woodpecker-go/woodpecker/const.go [hunk 1]
```diff
@@ -44,3 +44,14 @@ const (
 	LogEntryMetadata
 	LogEntryProgress
 )
+
+// LogEntryType identifies the type of step
```
Comment (6543) [inline woodpecker-go/woodpecker/const.go:48]:
```suggestion
// StepType identifies the type of step
```

File: pipeline/frontend/yaml/compiler/convert.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ import (
 	yaml_types "github.com/woodpecker-ci/woodpecker/pipeline/frontend/yaml/types"
 )
 
-func (c *Compiler) createProcess(name string, container *yaml_types.Container, section string) *backend_types.Step {
+func (c *Compiler) createProcess(name string, container *yaml_types.Container, typ backend_types.StepType) *backend_types.Step {
```
Comment (anbraten) [inline pipeline/frontend/yaml/compiler/convert.go:19]:
```suggestion
func (c *Compiler) createProcess(name string, container *yaml_types.Container, type backend_types.StepType) *backend_types.Step {
```

File: server/model/step.go [hunk 3]
```diff
@@ -67,3 +68,13 @@ func (p *Step) Running() bool {
 func (p *Step) Failing() bool {
 	return p.Failure == FailureFail && (p.State == StatusError || p.State == StatusKilled || p.State == StatusFailure)
 }
+
+type StepType uint8 //	@name StepType
+
+const (
```
Comment (anbraten) [inline server/model/step.go:74]: Could we re-use the pipeline model in here.

File: pipeline/frontend/yaml/compiler/convert.go (hunk 2 above)
Comment (6543) [reply pipeline/frontend/yaml/compiler/convert.go:19]: this was intentional to not overload the `type` keyword ... but I can rename it to stepType I guess

File: server/model/step.go (hunk 3 above)
Comment (6543) [reply server/model/step.go:74]: what do you want to reuse?

Comment (anbraten) [reply server/model/step.go:74]: The enum from pipeline

Comment (6543) [reply server/model/step.go:74]: Ah ... no because of imports

Comment (6543) [reply server/model/step.go:74]: e.g. just exec:
```sh
go list -f '{{.Imports}}' github.com/woodpecker-ci/woodpecker/pipeline/backend/... | tr -d '[]' | tr ' ' '\n' | grep 'githu
b.com/woodpecker-ci/woodpecker' | sort | uniq
```

File: pipeline/backend/types/step.go [hunk 4]
```diff
@@ -35,3 +36,13 @@ type Step struct {
...
+
+type StepType uint8
+
+const (
+	StepTypeClone StepType = iota + 1
+	StepTypeService
+	StepTypePlugin
+	StepTypeCommands
+	StepTypeCache
+)
```
[the remark this answers is not in the document]
Comment (6543) [reply pipeline/backend/types/step.go:48]: Sure, it just consumes more space ...

[the remark this answers is not in the document]
Comment (anbraten) [reply pipeline/backend/types/step.go:48]: Sure, but I think its okay and that way and much easier to understand for someone using the api

[the remark this answers is not in the document]
Comment (6543) [reply pipeline/backend/types/step.go:48]: 2min ... and it's changed

[the remark this answers is not in the document]
Comment (6543) [reply pipeline/backend/types/step.go:48]: @anbraten can you look at  # 1969 meanwhile?

**Pull request merged**